### PR TITLE
Clean up events, update liquidity pool docs

### PR DESCRIPTION
**Pull request by hieronx**

### CODE REVIEW

File: src/PoolManager.sol [hunk 1]
```diff
@@ -442,31 +440,21 @@ contract PoolManager is Auth {
...
     {
-        TrancheTokenPrice memory value = pools[poolId].tranches[trancheId].prices[currencyAddress];
+        TrancheTokenPrice memory value = pools[poolId].tranches[trancheId].prices[currency];
         price = value.price;
         computedAt = value.computedAt;
     }
 
-    function isAllowedAsInvestmentCurrency(uint64 poolId, address currencyAddress) public view returns (bool) {
-        uint128 currency = currencyAddressToId[currencyAddress];
-        if (currency == 0) {
```
Comment (hieronx) [inline src/PoolManager.sol:465]: This is already checked in `addCurrency`. And `allowInvestmentCurrency` can only be called after `addCurrency`.

**Review by NunoAlexandre — APPROVED** [review]
The changes look good to me. One question tho: is it an idea that we verify the emitted events in the tests? Not sure if that's an idiomatic thing to do in Solidity but I'd see value in having that safe-guard in place. wdyt?

**Comment by hieronx** [comment]
> One question tho: is it an idea that we verify the emitted events in the tests? Not sure if that's an idiomatic thing to do in Solidity but I'd see value in having that safe-guard in place. wdyt?

Yeah we definitely should add tests for this! We just haven't gotten to this yet. We can use https://book.getfoundry.sh/cheatcodes/expect-emit for this

**Comment by NunoAlexandre** [comment]
> > One question tho: is it an idea that we verify the emitted events in the tests? Not sure if that's an idiomatic thing to do in Solidity but I'd see value in having that safe-guard in place. wdyt?
> 
> Yeah we definitely should add tests for this! We just haven't gotten to this yet. We can use https://book.getfoundry.sh/cheatcodes/expect-emit for this

Nice, created an issue for that: https://github.com/centrifuge/liquidity-pools/issues/201

**Review by ilinzweilin — APPROVED** [review]

**Review by AStox — APPROVED** [review]
Looks good!

**Pull request merged**